samples/identity/credentials/src/default-credential-chain-debug-logger.ts

```diff
@@ -0,0 +1,66 @@
+// <CREDENTIAL_CHAIN_DEBUG
+import { setLogLevel, AzureLogger } from "@azure/logger";
+import { BlobServiceClient } from "@azure/storage-blob";
+import { 
+    ChainedTokenCredential, 
+    AzureDeveloperCliCredential,
+    AzureCliCredential 
+} from "@azure/identity";
+
+// Constant for the Azure identity log prefix
+const AZURE_IDENTITY_LOG_PREFIX = "azure:identity";
+
+// override logging to output to console.log (default location is stderr)
+// only log messages that start with the Azure identity log prefix
+setLogLevel("verbose");
+AzureLogger.log = (...args) => {
+  const message = args[0];
+  if (typeof message === 'string' && message.startsWith(AZURE_IDENTITY_LOG_PREFIX)) {
+    console.log(...args);
+  }
+};
+
+// Get storage account name from environment variable
+const storageAccountName = process.env.AZURE_STORAGE_ACCOUNT_NAME;
+
+if (!storageAccountName) {
+    throw new Error("AZURE_STORAGE_ACCOUNT_NAME environment variable is required");
+}
+
+const credential = new ChainedTokenCredential(

```

Comment:
I suggest using `DefaultAzureCredential` instead. That way, any reader using VS Code, Azure PowerShell, or a broker can still run the code successfully. Make sure you set env var `AZURE_TOKEN_CREDENTIALS` equal to "dev" when you switch to DAC.

Reply:
Changed 2 two files - one is DAC, on is a dev focused cred chain.